File: Programmers/Lv2/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    for tree in skill_trees:
        stack = []
        for w in tree:
            if w in skill:
                stack.append(w)
        flag = True
        for i in range(len(stack)):
            if stack[i] != skill[i]:
                flag = False
            if not flag:
                break
        if flag:
            answer += 1

    return answer

File: Programmers/Lv2/스킬트리.js
function solution(skill, skill_trees) {
    var answer = 0;
    
    for(tree of skill_trees){
        var stack = [];
        for(w of tree){
            if(skill.includes(w)){
                stack.push(w)
            }
        }
        var flag = true;
        for(let i = 0 ; i < stack.length; i++){
            if(stack[i] !== skill[i]){
                flag = false;
                break;
            }
        }
        if(flag){
            answer += 1;
        }
    
    }
    return answer;
}